src/main/java/org/terasology/dynamicCities/parcels/DynParcelTypeHandler.java

```diff
@@ -15,41 +15,46 @@
  */
 package org.terasology.dynamicCities.parcels;
 
-
 import com.google.common.collect.ImmutableMap;
 import org.terasology.commonworld.Orientation;
 import org.terasology.math.geom.Rect2i;
-import org.terasology.persistence.typeHandling.DeserializationContext;
 import org.terasology.persistence.typeHandling.PersistedData;
 import org.terasology.persistence.typeHandling.PersistedDataMap;
 import org.terasology.persistence.typeHandling.RegisterTypeHandler;
-import org.terasology.persistence.typeHandling.SerializationContext;
-import org.terasology.persistence.typeHandling.SimpleTypeHandler;
+import org.terasology.persistence.typeHandling.*;
 
 import java.util.Map;
+import java.util.Optional;
 
 @RegisterTypeHandler
-public class DynParcelTypeHandler extends SimpleTypeHandler<DynParcel> {
+public class DynParcelTypeHandler extends TypeHandler<DynParcel> {
 
     @Override
-    public PersistedData serialize(DynParcel parcel, SerializationContext context) {
+    public PersistedData serialize(DynParcel parcel, PersistedDataSerializer serializer) {
```

Comment:
Please move the code you have in this method to `serializeNonNull` and delete this method.